packages/bot-web-ui/src/components/dashboard/quick-strategy/quick-strategy-components/fields.tsx

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import classNames from 'classnames';
+import { SelectField, InputField, data_fields, data_uniq_input_obj } from '.';
+import { TDataFields } from './data/data-fields';
+import {
+    TSelectsFieldNames,
+    TDropdownItems,
+    TInputBaseFields,
+    TInputsFieldNames,
+    TDropdowns,
+    TSelectedValuesSelect,
+} from '../quick-strategy.types';
+import { TQStrategyFields } from './components.types';
+
+const QStrategyFields = React.memo(
+    ({
+        is_mobile,
+        types_strategies_dropdown,
+        symbol_dropdown,
+        trade_type_dropdown,
+        duration_unit_dropdown,
+        selected_type_strategy,
+        selected_trade_type,
+        selected_symbol,
+        selected_duration_unit,
+        onChangeDropdownItem,
+        onHideDropdownList,
+        setFieldValue,
+        onScrollStopDropdownList,
+        handleChange,
+        onChangeInputValue,
+        setCurrentFocus,
+        values,
+        description,
+        errors,
+    }: TQStrategyFields) => {
+        const is_double_idx_ref: React.MutableRefObject<number | undefined> = React.useRef();
+
+        const uniq_selected_input = React.useMemo(
+            () => data_uniq_input_obj.filter(elem => elem.index === selected_type_strategy.index)[0],
+            [selected_type_strategy]
+        );
+
+        const fields = React.useMemo(
+            () =>
+                data_fields.map((item, idx) => {
+                    const {
+                        id,
+                        name,
+                        field_name,
+                        className,
+                        label_className,
+                        field_className,
+                        input_value,
+                        label,
+                        placeholder,
+                        trailing_icon_message,
+                        zIndex,
+                        is_able_disabled,
+                    } = item as TDataFields;
+
+                    const is_uniq_strategy_field = Array.isArray(item);
+                    const is_input_field = is_uniq_strategy_field || input_value.startsWith('input_');
```

Comment:
to check if the field is an input or not, isn't it better to add a prop to the data?

Reply:
@sara-fs which line does this comment refer to?

Reply:
input_value.startsWith('input_');

Reply:
On the one hand yes, on the other hand, why not get rid of additional properties where possible?

Reply:
@sara-fs fixed